x/state-verifier/keeper/query.go

```diff
@@ -0,0 +1,18 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/neutron-org/neutron/v5/x/state-verifier/types"
+)
+
+// VerifyStateValues implements `VerifyStateValues` gRPC query to verify storage values
+func (k *Keeper) VerifyStateValues(ctx context.Context, request *types.QueryVerifyStateValuesRequest) (*types.QueryVerifyStateValuesResponse, error) {
+	if err := k.Verify(sdk.UnwrapSDKContext(ctx), int64(request.Height), request.StorageValues); err != nil {
+		return nil, err
+	}
+
+	return &types.QueryVerifyStateValuesResponse{Valid: true}, nil
```

Comment:
Yeah, but i don't have any others good ways in mind:
* returning just an empty struct seems not right as well;
* returning false (not an err) in case values are not proovable doesn't seem correct as well (we need return an err to tell what's wrong exactly);

Do you have any suggestions?

Reply:
Agreed, seems to be no other way